So I found that when I used the `-C` option I received this error:

```
module.js:340
    throw err;
    ^
Error: Cannot find module 'config.json'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/Users/anu01/bin/node-v0.10.26-darwin-x64/lib/node_modules/npm-lazy-mirror/lib/config.js:56:14)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
```

--- a/example/server-config.json
+++ b/example/server-config.json
@@ -22,6 +22,7 @@
     "upstream_port": "443",
     "upstream_use_https": true,
     "upstream_verify_ssl": true,
+    "upstream_db_path": "/",
 
     "cache_expiry": 320000,
     "cache_dir": "/tmp/npm-lazy",

--- a/lib/config.js
+++ b/lib/config.js
@@ -8,7 +8,7 @@ var Config = require('../config/config-defaults');
 var PackageJSON = require('../package');
 var Lactate = require('lactate');
 var Npm = require('npm');
-
+var FS = require('fs');
 
 var Optimist = require('optimist')
 .usage('Usage: $0 -p <port> -r <external_dns> -c <file cache dir>')
@@ -52,7 +52,7 @@ if (Argv.h) {
     process.exit();
 
 } else if (Argv.C || Argv.config) {
-    Config = require(Argv.C || Argv.config);
+    Config = JSON.parse(FS.readFileSync((Argv.C || Argv.config)));
 }
 
 /* Local server options */

--- a/lib/registry.js
+++ b/lib/registry.js
@@ -100,7 +100,7 @@ Registry.prototype.getMeta = function (pkg, cb) {
     var proto = this.config.upstreamUseHttps ? 'https' : 'http';
 
     var requestOpts = {
-        url: proto + '://' + this.config.upstreamHost + ':' + this.config.upstreamPort + this.config.upstreamDBPath + pkg,
+        url: proto + '://' + this.config.upstreamHost + ':' + this.config.upstreamPort + this.config.upstreamDBPath + '/' + pkg,
         strictSSL: this.config.upstreamVerifySSL
     };
 
@@ -146,7 +146,7 @@ Registry.prototype.getTarball = function (pkg, cb) {
         },
         function (done) {
             var requestOpts = {
-                url: proto + '://' + pkg.config.upstreamHost + ':' + this.config.upstreamPort + this.config.upstreamDBPath + pkg.name + '/-/' + pkg.name + '-' + pkg.version + '.tgz',
+                url: proto + '://' + pkg.config.upstreamHost + ':' + this.config.upstreamPort + this.config.upstreamDBPath + '/' + pkg.name + '/-/' + pkg.name + '-' + pkg.version + '.tgz',
                 strictSSL: registry.config.upstreamVerifySSL
             };
 

